azavea/iow-boundary-tool
# Add Prototype Skeleton

## Overview

This adds 2 base pages: http://localhost:4545/welcome and http://localhost:4545/main which correspond to the Welcome Modal and Map and Sidebar pages respectively, allowing them to be developed independently.

The Welcome Model has a Leaflet Map using [ESRI's Raster Light Gray Canvas basemap](https://www.arcgis.com/home/item.html?id=8b3d38c0819547faa83f7b7aca80bd76) centered to North Carolina in the background, with a non-dismissable [Chakra UI Modal](https://chakra-ui.com/docs/components/modal) in the foreground. This corresponds to [the Welcome Flow wireframes](https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=168%3A2700) on Figma.

The Map and Sidebar page uses the same map, with a sidebar. This layout uses [Chakra's Flex component](https://chakra-ui.com/docs/components/flex).

Once this is in, we should make cards for the individual sub-pages. A reasonable attempt should be made to match the wireframes, in terms of colors / spacing / fonts.

Thanks to @mstone121, @jacobtylerwalls, and @rachelekm for their input in this work.

Closes #15 

### Demo

![image](https://user-images.githubusercontent.com/1430060/183905990-5604678c-2f29-4bf9-9be4-dd6d8badff36.png)

![image](https://user-images.githubusercontent.com/1430060/183906024-06e85e3d-de93-4319-8607-66ecc1b17566.png)

### Notes

Seeing this warning in the Webpack output, will address this in a future PR:

```
WARNING in ./node_modules/leaflet/dist/leaflet.css (./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[5].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[5].use[2]!./node_modules/source-map-loader/dist/cjs.js!./node_modules/leaflet/dist/leaflet.css)
Module Warning (from ./node_modules/postcss-loader/dist/cjs.js):
Warning

(655:3) autoprefixer: Replace color-adjust to print-color-adjust. The color-adjust shorthand is currently deprecated.

webpack compiled with 1 warning
Compiling...
Compiled with warnings.

Warning
(655:3) autoprefixer: Replace color-adjust to print-color-adjust. The color-adjust shorthand is currently deprecated.

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before
```

## Testing Instructions

- Check out this branch and `update` and `server`
- Go to http://localhost:4545/
  - [ ] Ensure you are redirected to http://localhost:4545/welcome
  - [ ] Ensure you see the grayscale map in the background, with a modal on top
- Go to http://localhost:4545/main
  - [ ] Ensure you see a blue-gray sidebar next to the map
- Inspect the JavaScript console
  - [ ] Ensure there are no errors
